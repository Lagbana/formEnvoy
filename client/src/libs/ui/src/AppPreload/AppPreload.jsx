import React, { useEffect } from 'react';
import { useState } from 'react';
import * as Font from 'expo-font';
import * as SplashScreen from 'expo-splash-screen'
import * as Sentry from 'sentry-expo';
import { Maybe } from 'colibri/common';
import { Text } from 'react-native';

export type AppFontsType = string | { [fontFamily: string]: Font.FontSource }

export interface PreloadProviderProps {
	fonts?: AppFontsType;
}
export const AppPreload: React.FC<PreloadProviderProps> = ({children, fonts = {}}) => {

	const [initialized, setInitialized] = useState(false);
	const [error, setError] = useState<Maybe<string>>();

	useEffect(() => {
		startAsync().then(
			SplashScreen.hideAsync
		).then(
			() => setInitialized(true)
		)
		.catch((err) => {
			console.log(err);
			Sentry.captureMessage(JSON.stringify(err));
			setError(err);
		});

	}, []);


	const startAsync = () => Font.loadAsync(fonts);
	
	// This is the main app boot screen which is different from the
	// standard colibri/ui AppLoading component, so should use the expo version

	if (error) {
		return <Text>{JSON.stringify(error)}</Text>;
	} else if (!initialized) {
		return null;
	} else {
		return <>{children}</>;
	}

}