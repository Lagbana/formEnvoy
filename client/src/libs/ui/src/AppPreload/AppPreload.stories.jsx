import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { AppPreload } from './AppPreload'

storiesOf('AppPreload', module).add('Default', () => (
  <AppPreload children={'Application loading ...'} />
))
