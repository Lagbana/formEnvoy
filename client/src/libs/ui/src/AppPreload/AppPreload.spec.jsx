import React from 'react'
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { AppPreload } from './AppPreload'

configure({ adapter: new Adapter() })

test('<AppPreload /> renders correctly', () => {
  const component = shallow(<AppPreload />)
  component.unmount()
  expect(toJson(component)).toMatchSnapshot()
})

