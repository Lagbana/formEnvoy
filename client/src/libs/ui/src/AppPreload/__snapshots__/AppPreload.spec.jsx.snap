// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppPreload /> renders correctly 1`] = `""`;
